internal/globalaccounts/globalaccounts.go

```diff
@@ -253,54 +138,100 @@ func svcRequest(config Config, svc *http.Client, subaccountId string, logs *logr
 		}
 	}()
 	if response.StatusCode != http.StatusOK {
-		return result{}, fmt.Errorf("url: %s : response status -> %s", request.URL, response.Status)
+		return svcResult{}, fmt.Errorf("while fail on url: %s : due to response status -> %s", request.URL, response.Status)
 	}
-	var svcResponse result
+	var svcResponse svcResult
 	err = json.NewDecoder(response.Body).Decode(&svcResponse)
 	if err != nil {
 		logs.Errorf("while decoding response: %s", err.Error())
-		return result{}, err
+		return svcResult{}, err
 	}
 	return svcResponse, nil
 }
 
-func fixGlobalAccounts(db storage.Instances, kcp client.Client, cfg Config, toFix []fixMap, logs *logrus.Logger) {
-	_ = broker.NewLabeler(kcp)
-	updateErrorCounts := 0
-	processed := 0
-	logs.Infof("fix start. Is dry run?: %t", cfg.DryRun)
-	for _, pair := range toFix {
-		processed++
-		if cfg.DryRun {
-			logs.Infof("dry run: update labels for runtime %s with new %s", pair.instance.RuntimeID, pair.correctGlobalAccountId)
+func logic(config Config, svc *http.Client, kcp client.Client, connection *dbr.Connection, db storage.BrokerStorage, logs *logrus.Logger) {
+	var okCount, getInstanceErrorCounts, requestErrorCount, mismatch, kebInstanceMissingSACount, kebInstanceMissingGACount int
+	var instanceUpdateErrorCount, labelsUpdateErrorCount int
+	var out strings.Builder
+	labeler := broker.NewLabeler(kcp)
+
+	instancesIDs := make([]string, 0)
+	_ = connection.QueryRow("select instance_id from instances").Scan(&instancesIDs) // suspended ones
+	for i, instanceID := range instancesIDs {
```

Comment:
We have very long loop body. We could extract it to separate function processing given instance.